packages/branch-names/src/branches.ts

```diff
@@ -0,0 +1,63 @@
+import { GitHubActionsContext } from "@octopusdeploy/shared-action-utils";
+
+export function getBranchNames(context: GitHubActionsContext) {
+    try {
+        let isDefault: boolean = false;
+        let isTag: boolean = false;
+        let defaultBranch: string = "";
+        let currentBranch: string = "";
+        let baseRefBranch: string = "";
+        let headRefBranch: string = "";
+        let refBranch: string = "";
+        let tag: string = "";
+
+        const githubRef = context.getInput("gitHubRef");
+        const githubBaseRef = context.getInput("gitHubBaseRef");
+        const githubHeadRef = context.getInput("gitHubHeadRef");
+        const githubEventBaseRef = context.getInput("gitHubEventBaseRef");
+        const githubEventName = context.getInput("gitHubEventName");
+        const githubEventRepositoryDefaultBranch = context.getInput("gitHubEventRepositoryDefaultBranch");
+        const tagPrefixToRemove = context.getInput("tagPrefixToRemove");
+
+        if (!githubRef.startsWith("refs/tags/")) {
+            baseRefBranch = githubBaseRef.replace(/refs\/heads\//, "");
+            headRefBranch = githubHeadRef.replace(/refs\/heads\//, "");
+            refBranch = githubRef.replace(/refs\/pull\//, "").replace(/refs\/heads\//, "");
+
+            if (githubEventName === "pull_request") {
+                currentBranch = headRefBranch;
+            } else {
+                currentBranch = refBranch;
+            }
+
+            defaultBranch = githubEventRepositoryDefaultBranch;
+            isDefault = currentBranch === githubEventRepositoryDefaultBranch;
+        } else {
+            const tagHeadRegex = new RegExp(`refs/heads/${tagPrefixToRemove}`, "i");
+            const tagRegex = new RegExp(`refs/tags/${tagPrefixToRemove}`, "i");
+            baseRefBranch = githubEventBaseRef.replace(tagHeadRegex, "");
+            tag = githubRef.replace(tagRegex, "");
+
+            baseRefBranch = baseRefBranch.replace(/refs\/heads\//, "");
+            tag = tag.replace(/refs\/tags\//, "");
+
+            isTag = true;
+        }
+
+        currentBranch = currentBranch.replaceAll(";", "-");
```

Comment:
Is this enough to make the branch names safe? I wonder how we can tell.

Also, I wonder if its worth making the replacement character configurable somehow. We could default it to something reasonable like we have here (`-`), or maybe even just an empty string?

Reply:
I have been trying to see if I can get a list of characters that should be removed, and I can't find a definitive list so not sure I will make the replacement character configurable though.

Reply:
I have decided that we should now allow this to be configured, as it might allow users to replace values we remove with other bad values. It just removes the characters, I have decided to remove these characters ";" and "#"